### DISCUSSION TITLE
Geoscan X-band

### DISCUSSION THREAD

**Discussion by cyyyr** [daniestevez/gr-satellites#450] [Satellite teams]
Geoscan X-band transmitter will be placed on radio amateur cubesats and will follow amateur missions, like:

- Translate a Cubesat's ham-radio callsign
- Transmit CW, photos and videos with GFSK modulation to radio-amateurs around the World
- Will be used for amateurs to receive data in X-band on request through Geoscan Website and libre.space forum

Following information will be needed to demodulate a transmission:

- Modulation - GFSK
- Syncword - 0xF68D
- Preambule - fourty 1’s (1111111111111111111111111111111111111111)
- Fixed size 64 byte data-packet inside radio-packet
- [Link ]( 
 https://drive.google.com/file/d/1k2jgX_4XjN0HmegTojzfD2B_F-5Lwxwk/view?usp=sharing) with example IQ files (sample rate 2048000).
 
 Info about our transmitter:

-   frequency - 10450-10500 MHz
-   baudrate - 125kbs
-   bandwifth is around 250 kHz

**Comment by daniestevez** [comment]
Thanks for sharing this information. A few questions:

- Is this X-band transmitter available and usable in GEOSCAN-EDELVEIS, or will it only be available in future GEOSCAN satellites?
- Are the specifications of the transmitter already frozen or still open to change?
- Do the packets use any FEC, scrambling and/or CRC?

One remark is that the preamble doesn't serve to train the symbol synchronization loop, because it has no symbol transitions. Therefore, the closed-loop FSK demodulator in gr-satellites probably won't work well with these frames.

**Reply by cyyyr** [reply]
Here's answers:

Is this X-band transmitter available and usable in GEOSCAN-EDELVEIS, or will it only be available in future GEOSCAN satellites?
- Yes, we have X-band transmitter available and it's alive, but we turned it off because of huge hardware frequency shift. Fully working transmitter will be available in future (with all possible uplink upgrades). First one will be launch this summer...
Are the specifications of the transmitter already frozen or still open to change?
- ... so there is not much time left but we still can change some things.
- We use only CRC (It should be CRC-16 or CRC-32, i'll specify bit a later)

 Hmm. Could you please advice, how to resolve this? Should we change preamble?

**Reply by daniestevez** [reply]
Looking at your recordings, the preamble doesn't fit what you described. Here is a screenshot of `SDRSharp_20230404_121646Z_698090000Hz_IQ.wav` in inspectrum. The preamble is 010101..., or 101010...

![xband](https://user-images.githubusercontent.com/15093841/234110372-d783e592-bb89-4b61-9046-5dd619d0599c.png)

Is perhaps this caused by differential coding?

**Reply by cyyyr** [reply]
Sorry, I confused x with our uhf somehow:) 
We have FEC and scrambling. And we don't have CRC on x-band

**Reply by daniestevez** [reply]
Can you describe the FEC and scrambling used in X-band?

**Reply by cyyyr** [reply]
We have 1/2 FEC (Convolutional code) - 64 bytes after data-packet. And multiplicational scrambler

**Reply by daniestevez** [reply]
Can you give the full details for these? A document of several pages with all the definitions would be helpful.

**Reply by cyyyr** [reply]
Sure, I'll send it here as it will be ready! Thank you!

**Reply by cyyyr** [reply]
Hello, @daniestevez! Excuse me for a long reply, i've been away a few days! Here is our radio protocol document. 
[geoscan-x-trasmitter.odt](https://github.com/daniestevez/gr-satellites/files/11454255/geoscan-x-trasmitter.odt)

**Reply by daniestevez** [reply]
Hi @cyyyr, I'm afraid that documentation is very brief and ambiguous. I spotted at least the following problems:
- I am looking at IEEE 802.15.4-2020 Section 12: O-QPSK PHY, and I don't see any mention of convolutional coding. Section 21.3.6 SUN O-QPSK PHY; FEC describes a convolutional encoder, but it doesn't match the parameters you give (the constraint length is 7, not 5, for instance).
- Interleaving is not defined.
- Giving polynomials in hex or as polynomials can result ambiguous regarding bit endianness and bit order of LFSR.
- The scrambling parameters are somewhat ambiguous (it is not clear what output bit 0 means, for instance).
- The CRC parameters are insufficient to specify the code. There should be more endinanness parameters, as well as final XOR.

I don't want to spend more time iterating and pointing out what is missing in the documentation, so I would suggest that if you want to get support from the community to implement a decoder for this, you share all the documentation you have. In any case, I'm also happy to accept a pull request for the decoder if you implement it yourselves.

**Reply by cyyyr** [reply]
Sorry, if I did something wrong. We use high level functions to set transmitter up, so I am trying to figure out all nuances myself. It is not about that i am trying to hide something. I will provide any information that i could find and will help process as much as needed. 
I will do my best to make documentation less ambiguous for community. Thank you very much! And again I didn't mean to take much of your time.
